skmorf/tree/_unsup_criterion.pyx

```diff
@@ -133,35 +208,223 @@ cdef class TwoMeans(Criterion):
+                mu_right += X[i] / self.n_samples
+
+        for p in range(end):
+            i = self.sample_indices[p]
+
+            if self.sample_weight is not None:
+                w = self.sample_weight[i]
+
+            if k < pos:
+                left_c += w * (X[i] - mu_left)**2
+            else:
+                right_c += w * (X[i] - mu_right)**2
+
+        return (left_c + right_c) / self.n_samples
 
     cdef void children_impurity(self, double* impurity_left,
-                                double* impurity_right) nogil:
-        # need to implement
+                                    double* impurity_right) nogil:
+       """Evaluate the impurity in children nodes.
+
+        i.e. the impurity of the left child (samples[start:pos]) and the
+        impurity the right child (samples[pos:end]).
+
+        Parameters
+        ----------
+        impurity_left : double pointer
+            The memory address to save the impurity of the left node
+        impurity_right : double pointer
+            The memory address to save the impurity of the right node
+        """
+        # cdef double entropy_left = 0.0
+        # cdef double entropy_right = 0.0
+        # cdef double count_k
+        # cdef SIZE_t k
+        # cdef SIZE_t c
+
+        # for k in range(self.n_outputs):
+        #     for c in range(self.n_classes[k]):
+        #         count_k = self.sum_left[k, c]
+        #         if count_k > 0.0:
+        #             count_k /= self.weighted_n_left
+        #             entropy_left -= count_k * log(count_k)
+
+        #         count_k = self.sum_right[k, c]
+        #         if count_k > 0.0:
+        #             count_k /= self.weighted_n_right
+        #             entropy_right -= count_k * log(count_k)
+
+        # impurity_left[0] = entropy_left / self.n_outputs
+        # impurity_right[0] = entropy_right / self.n_outputs
+        
+        # use left_sum and right_sum to calculate impurity?
+        # or should this method take two separate Xs making up
+        # left and right child nodes?
         pass
 
     cdef void node_value(self, double* dest) nogil:
-        """Compute the node value of samples[start:end] and save it into dest.
+        r"""Compute the node value of samples[start:end] and save it into dest.
+
+        TODO: No, it does not handle the computation. 
+        See scikit-learn splitter and tree for how node_value() is used.
 
         Parameters
         ----------
         dest : double pointer
             The memory address which we will save the node value into.
         """
-        # need to implement
-        pass
+        cdef SIZE_t k
+
+        # TODO: no notion of classes
+        # Possible tip: See how node_value is used in splitter and tree.
+        for k in range(self.n_outputs):
+            memcpy(dest, &self.sum_total[k], sizeof(double))
+            # TODO: need to revisit
+            # dest += self.sum_total[k] / self.weighted_n_node_samples
```

Comment:
You should be able to fill in this implementation. It's a one-liner to set the value of `dest`

Reply:
I set `dest` to sum_total which doesn't need to be deferenced anymore

Reply:
I just removed all the n_outputs in the code so that this becomes a one line of code copying sum_total to dest